weave/trace/op.py

```diff
@@ -82,6 +82,8 @@ def print_call_link(call: "Call") -> None:
 
 FinishCallbackType = Callable[[Any, Optional[BaseException]], None]
 OnOutputHandlerType = Callable[[Any, FinishCallbackType, Dict], Any]
+# Call, original function output, exception if occurred
+OnFinishHandlerType = Callable[["Call", Any, Optional[BaseException]], None]
```

Comment:
Curious why `_on_output_handler` was insufficient here?

Reply:
It only gets the output value, the finish handler needs the whole call - in the Gemini case the handler is used to modify the summary object, and what it needs to set in the summary depends on parsing information (the model name) out of the inputs.